internal/api/v1/application/stage.go

```diff
@@ -360,14 +360,65 @@ func newJobRun(app stageParam) (*batchv1.Job, *corev1.Secret) {
 	// TODO: Extract scripts to config map -- https://github.com/epinio/epinio/issues/1175
 	// TODO: Simplify env setup -- https://github.com/epinio/epinio/issues/1176
 
-	awsScript := fmt.Sprintf("echo Extracting _ _ __ ___ _____ $(whoami) ; aws --endpoint-url %s://%s s3 cp s3://%s/%s /workspace/source/%s ; echo _ _ __ ___ _____ Extracted",
-		protocol, app.S3ConnectionDetails.Endpoint, app.S3ConnectionDetails.Bucket,
-		app.BlobUID, app.BlobUID)
-
-	unpackScript := fmt.Sprintf("echo Unpacking _ _ __ ___ _____ $(whoami) ; mkdir /workspace/source/app ; tar -xvf /workspace/source/%s -C /workspace/source/app/; rm /workspace/source/%s; chown -R 1000:1000 /workspace ; cp -vL /platform/appenv/* /platform/env/ ; ls -la /platform/env ; echo _ _ __ ___ _____ Unpacked", app.BlobUID, app.BlobUID)
-
-	buildpackScript := fmt.Sprintf(`trap "curl -X POST http://localhost:4191/shutdown || true" EXIT ; echo Creating _ _ __ ___ _____ $(whoami) ; ls -la /platform/env ; /cnb/lifecycle/creator -app=/workspace/source/app -cache-dir=/workspace/cache -uid=1000 -gid=1000 -layers=/layers -platform=/platform -report=/layers/report.toml -process-type=web -skip-restore=false -previous-image=%s %s`,
-		previous.ImageURL(previous.RegistryURL), app.ImageURL(app.RegistryURL))
+	// runtime: AWSCLIImage
+	awsScript := fmt.Sprintf(`echo By _ _ __ ___ _____ $(whoami)
+aws --endpoint-url %s://%s s3 cp s3://%s/%[4]s /workspace/source/%[4]s
+echo _ _ __ ___ _____ Done`,
+		protocol,
+		app.S3ConnectionDetails.Endpoint,
+		app.S3ConnectionDetails.Bucket,
+		app.BlobUID)
+
+	// runtime: BashImage
+	// Attempting to unpack the sources as, in order:
+	//    .tar	- epinio cli
+	//    .zip	- epinio UI
+	// -z .tar.gz
+	// -j .tar.bz2
+	// -J .tar.xz
+	//
+	// __Note__: While it would have been nicer, IMNSHO, to use
+	// `file` to determine the type of the file and then directly
+	// dispatch to the proper unpacker, the `file` command is not
+	// available in the `bash` image. The code as written now
+	// relies on each unpacker to recognize/reject input properly.
+
+	unpackScript := fmt.Sprintf(`echo By _ _ __ ___ _____ $(whoami)
+mkdir /workspace/source/app
+(  cd /workspace/source/app
+   ( echo Tar? ; tar -xvf  ../%[1]s | sed -e 's/^/Tar: /' ) || \
```

Comment:
Trying this `sed` locally, it doesn't seem to do anything when `tar` fails. That's because the output goes to stderr:

```
~ $ tar -xvf nottarfile | sed -e 's/^/Tar: /'
tar: This does not look like a tar archive
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
```
merging the outputs makes it work? (`tar -xvf nottar 2>&1`). Do we need the `sed` at all?

Reply:
The sed was added to get a good indication what format was picked, as part of the output from the chosen unarchiver.
I can certainly ensure to direct stderr through it also.

Reply:
New PR: #1197